infiniband-diags/ibdiag_sa.c

```diff
@@ -39,6 +39,7 @@
 
 #include <errno.h>
 #include <infiniband/umad.h>
+#include "../libibmad/mad_internal.h"
```

Comment:
Something wrong if you need to provide such path. You should handle it as part of CMakeList.txt.

mad_internal.h even in the name says that it shouldn't be used outside of libibmad